src/imars3d/backend/dataio/data.py

```diff
@@ -475,41 +475,131 @@ def _extract_rotation_angles(
+    name: str
+        Used to name file of output, defaults to ``save_data``
+    omegas: Array
+        Optional for writing out the array of omega angles
+
+    Returns
+    -------
+        The directory the files were actually saved in
+    """
+
+    data = param.Array(doc="Data to save", precedence=1)
+    outputbase = param.Foldername(default="/tmp/", doc="radiograph directory")
+    name = param.String(default="save_data", doc="name for the radiograph")
+    omegas = param.Array(doc="Collection of omega angles")
+
+    def __call__(self, **params):
+        """Parse inputs and perform multiple dispatch."""
+        # type bounds check via Parameter
+        with param.logging_level("CRITICAL"):
+            # do not complain about directories that don't exist
+            _ = self.instance(**params)
+        # sanitize arguments
+        params = param.ParamOverrides(self, params)
+
+        if params.data is None:
+            raise ValueError("Did not supply data")
+
+        save_dir = params.outputbase / f"{params.name}_{_to_time_str(datetime.now())}"
+
+        # save the data as tiffs
+        _save_data(filename=save_dir / params.name, data=params.data, omegas=params.omegas)
+
+        return save_dir
+
+
+class save_checkpoint(param.ParameterizedFunction):
+    """
+    Save current state to checkpoint in a datetime stamped directory name.
+
+    The filenames will be
+    ``<outputbase>/<name>_chkpt_YYYYMMDDhhmm/<name>_####.tiff``
+    where a canonical ``outputbase`` is ``/HFIR/CG1D/IPTS-23788/shared/processed_data/``.
+
     Parameters
     ----------
     data: Array
         array of data to save
-    outputdir: str
-        where to save the output on disk
-    filename: str
+    outputbase: Path
+        The parent directory of where to save the output on disk.
+        ``param.Foldername`` will warn if the directory does not already exist.
+    name: str
         Used to name file of output, defaults to output_{datetime}
+    omegas: Array
+        Optional for writing out the array of omega angles
 
     Returns
     -------
-        None
+        The directory the files were actually saved in
     """
 
-    #
     data = param.Array(doc="Data to save", precedence=1)
-    outputdir = param.Foldername(default="/tmp/", doc="radiograph directory")
+    outputbase = param.Foldername(default="/tmp/", doc="directory checkpoint should exist in")
 
-    filename = param.String(default="*", doc="fnmatch for selecting dc files from dc_dir")
+    name = param.String(default="*", doc="name for the checkpoint")
+    omegas = param.Array(doc="Collection of omega angles")
 
     def __call__(self, **params):
         """Parse inputs and perform multiple dispatch."""
         # type bounds check via Parameter
-        _ = self.instance(**params)
+        with param.logging_level("CRITICAL"):
+            # do not complain about directories that don't exist
+            _ = self.instance(**params)
```

Comment:
should we do this for all of our param checking?  I think we only cares about type error and bounds error, and not so much about warnings.